todo/src/reducers/index.js

```diff
@@ -0,0 +1,32 @@
+import { ADD_TODO, TOGGLE_TODO } from '../actions';
+
+export default (state = [], action) => {
+  switch (action.type) {
+  case ADD_TODO:
+    return [...state, action.payload];
+
+  case TOGGLE_TODO:
+    return state.map((todo) => {
+      if (action.payload === todo.id) {
+	todo.complete = !todo.complete;
+      return todo;
+	 
+      }
+      else {
+	return todo;
+      }
+    });
+
+    default:
+     return state;
+
+  }
+};
+
+
+//So it seems that here in the reducer, we refer back to the actions we created. They were sent to the store and now the reducer is telling it how to handle those actions.
+	//Most of the examples I've seen use switch (action.type) in the reducer. I suppose this is like an if/then sort of thing. Like in this case return state and payload and in that case figure out whether the todo is complete or not and act accordingly. 
+	//Since the reducer always has to return a state, you need a default. Just in case.
+
```

Comment:
Yes! Reducer does the thing. A switch case is logically like an if/else, you're right. You don't really need it here, because you've only got one case. A good time to use switch would be if you had a Toggle function _and_ a delete function and you want your app to do one or the other. The app will read the requirements for each case and then perform the correct task based on the requirements.